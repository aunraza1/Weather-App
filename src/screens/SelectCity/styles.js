import {StyleSheet} from 'react-native';
import {vh, vw} from '../../utils/units';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    marginTop: vh * 3,
    paddingHorizontal: vw * 5,
  },
  heading: {
    alignSelf: 'center',
    fontSize: vh * 2.5,
  },
  subHeading: {
    marginTop: vh,
  },
});
export default styles;
