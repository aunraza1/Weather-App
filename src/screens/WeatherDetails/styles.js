import {StyleSheet} from 'react-native';
import {vh, vw} from '../../utils/units';
import {colors} from '../../utils/theme';

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  centeredView: {
    justifyContent: 'center',
    alignItems: 'center',
  },
  row: {
    marginTop: vh * 3,
    alignItems: 'center',
    flexDirection: 'row',
  },
  bullet: {
    backgroundColor: colors.orange,
    borderRadius: (vw * 2) / 2,
    width: vw * 2,
    height: vw * 2,
  },
  marginLeft: {
    marginLeft: vw * 2,
  },
  icon: {
    tintColor: colors.white,
    width: vw * 30,
    height: vh * 15,
  },
  temprature: {
    fontSize: vh * 3,
    alignSelf: 'center',
  },
  descriptionView: {
    marginTop: vh * 2,
    backgroundColor: colors.navyBlue,
    borderRadius: vw * 3,
    justifyContent: 'center',
    alignItems: 'center',
    paddingVertical: vh,
    paddingHorizontal: vw * 3,
    maxWidth: vw * 37,
  },
  boxView: {
    marginTop: vh * 2,

    width: '32%',
    marginRight: '1%',
  },
  boxImage: {
    tintColor: colors.white,
    resizeMode: 'contain',
    width: '100%',
    height: vh * 4,
  },
  value: {
    marginTop: vh,
    alignSelf: 'center',
  },
});
export default styles;
