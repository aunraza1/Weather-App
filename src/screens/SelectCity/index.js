import {ScrollView, View} from 'react-native';
import React from 'react';
import styles from './styles';
import {BoldText, RegularText, ValuePicker, Wrapper} from '../../components';
import useSelectCityController from '../../view-controllers/useSelectCityController';
import Dropdown from '../../components/DropDown';
import {cities} from '../../config';

const SelectCity = () => {
  const {showDropDown, handleDropDown, onSelectCity} =
    useSelectCityController();
  return (
    <Wrapper>
      <ScrollView style={styles.container}>
        <BoldText style={styles.heading} text={'Select Location'} />
        <RegularText
          style={styles.subHeading}
          numberOfLines={2}
          text={'Select the city you are looking for knowing the weather'}
        />
        <ValuePicker onPress={() => handleDropDown(false)} />
      </ScrollView>
      <Dropdown
        label={'Cities'}
        items={cities}
        onClose={() => handleDropDown(true)}
        onPressItem={name => {
          onSelectCity(name);
          handleDropDown(true);
        }}
        visible={showDropDown}
      />
    </Wrapper>
  );
};
export default SelectCity;
