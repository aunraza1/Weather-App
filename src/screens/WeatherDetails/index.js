import React from 'react';
import {
  BoldText,
  Header,
  LightText,
  Loader,
  RegularText,
  Wrapper,
} from '../../components';
import useWeatherDetailController from '../../view-controllers/useWeatherDetailsController';
import {ScrollView, View, Image} from 'react-native';
import styles from './styles';
import moment from 'moment';
import {icons} from '../../assets';
import {weatherIcons} from '../../config';

const WeatherDetails = ({route}) => {
  const city = route?.params?.city;
  const {details, loading, onPressGoBack} = useWeatherDetailController(city);

  const extraDetails = [
    {
      id: 1,
      name: 'Humidity',
      value: details?.main?.humidity + '%',
      icon: icons.humidity,
    },
    {
      id: 2,
      name: 'Feels Like',
      value: Math.ceil(details?.main?.feels_like) + '°C',
      icon: icons.temprature,
    },
    {
      id: 3,
      name: 'Wind Speed',
      value: details?.wind?.speed + 'm/s',
      icon: icons.wind,
    },
  ];
  return (
    <Wrapper>
      <Header onPress={onPressGoBack} title={city} />
      {loading ? (
        <Loader />
      ) : (
        <ScrollView style={styles.container}>
          <View style={styles.centeredView}>
            <View style={styles.row}>
              <View style={styles.bullet} />
              <LightText
                style={styles.marginLeft}
                text={`Today  ${moment(new Date()).format('hh:mm')}`}
              />
            </View>
            <Image
              style={styles.icon}
              source={
                weatherIcons[details?.weather[0]?.main]
                  ? weatherIcons[details?.weather[0]?.main]
                  : icons.clouds
              }
            />
            <BoldText
              style={styles.temprature}
              text={`${Math.ceil(details?.main?.temp)} °C`}
            />
            <View style={styles.descriptionView}>
              <RegularText text={details?.weather[0]?.description} />
            </View>
            <View style={[styles.row, {width: '100%'}]}>
              {extraDetails?.map((v, i) => (
                <View key={i} style={styles.boxView}>
                  <Image style={styles.boxImage} source={v?.icon} />
                  <BoldText style={styles.value} text={v?.value} />
                  <RegularText style={styles.value} text={v?.name} />
                </View>
              ))}
            </View>
          </View>
        </ScrollView>
      )}
    </Wrapper>
  );
};
export default WeatherDetails;
