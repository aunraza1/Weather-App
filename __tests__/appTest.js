import React from 'react';
import renderer from 'react-test-renderer';
import SelectCity from '../src/screens/SelectCity';
import WeatherDetails from '../src/screens/WeatherDetails';


jest.mock('@react-navigation/native', () => ({
    useNavigation: jest.fn(),
    useFocusEffect:jest.fn()
  }));
test('renders correctly', () => {
  const tree = renderer.create(<SelectCity />).toJSON();
  expect(tree).toMatchSnapshot();
});
test('renders correctly', () => {
  const tree = renderer.create(<WeatherDetails />).toJSON();
  expect(tree).toMatchSnapshot();
});