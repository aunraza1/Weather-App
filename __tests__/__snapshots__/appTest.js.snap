// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#0B0B45",
        "flex": 1,
        "paddingHorizontal": 37.5,
        "paddingVertical": 26.68,
      },
      undefined,
    ]
  }
>
  <RCTScrollView
    style={
      {
        "flex": 1,
        "marginTop": 40.019999999999996,
        "paddingHorizontal": 37.5,
      }
    }
  >
    <View>
      <Text
        allowFontScaling={false}
        style={
          [
            {
              "alignSelf": "flex-start",
              "color": "#FFFFFF",
              "fontFamily": "Montserrat-Bold",
              "fontSize": 20.009999999999998,
            },
            {
              "alignSelf": "center",
              "fontSize": 33.35,
            },
          ]
        }
      >
        Select Location
      </Text>
      <Text
        allowFontScaling={false}
        numberOfLines={2}
        style={
          [
            {
              "color": "#FFFFFF",
              "fontFamily": "Montserrat-Regular",
              "fontSize": 20.009999999999998,
            },
            {
              "marginTop": 13.34,
            },
          ]
        }
      >
        Select the city you are looking for knowing the weather
      </Text>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "backgroundColor": "#28356b",
            "borderRadius": 45,
            "flexDirection": "row",
            "height": 80.03999999999999,
            "marginTop": 26.68,
            "opacity": 1,
            "padding": 6.67,
            "width": "100%",
          }
        }
      >
        <View
          style={
            {
              "height": "100%",
              "justifyContent": "center",
              "paddingLeft": 30,
              "width": "80%",
            }
          }
        >
          <Text
            allowFontScaling={false}
            numberOfLines={1}
            style={
              [
                {
                  "color": "#FFFFFF",
                  "fontFamily": "Montserrat-Regular",
                  "fontSize": 20.009999999999998,
                },
                undefined,
              ]
            }
          >
            Select Location
          </Text>
        </View>
        <View
          style={
            {
              "alignItems": "center",
              "height": "100%",
              "width": "20%",
            }
          }
        >
          <Image
            source={
              {
                "testUri": "../../../src/assets/icons/dropdown.png",
              }
            }
            style={
              {
                "height": "100%",
                "resizeMode": "contain",
                "tintColor": "#FFFFFF",
                "width": "60%",
              }
            }
          />
        </View>
      </View>
    </View>
  </RCTScrollView>
  <Modal
    hardwareAccelerated={false}
    statusBarTranslucent={true}
    transparent={true}
    visible={false}
  />
</View>
`;

exports[`renders correctly 2`] = `
<View
  style={
    [
      {
        "backgroundColor": "#0B0B45",
        "flex": 1,
        "paddingHorizontal": 37.5,
        "paddingVertical": 26.68,
      },
      undefined,
    ]
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "marginTop": 26.68,
        "width": "100%",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
          "width": "20%",
        }
      }
    >
      <Image
        source={
          {
            "testUri": "../../../src/assets/icons/backArrow.png",
          }
        }
        style={
          {
            "height": 40.019999999999996,
            "resizeMode": "contain",
            "tintColor": "#FFFFFF",
            "width": "80%",
          }
        }
      />
    </View>
    <Text
      allowFontScaling={false}
      style={
        [
          {
            "alignSelf": "flex-start",
            "color": "#FFFFFF",
            "fontFamily": "Montserrat-Bold",
            "fontSize": 20.009999999999998,
          },
          {
            "fontSize": 33.35,
            "marginLeft": 112.5,
          },
        ]
      }
    />
  </View>
  <RCTScrollView
    style={
      {
        "flex": 1,
      }
    }
  >
    <View>
      <View
        style={
          {
            "alignItems": "center",
            "justifyContent": "center",
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
              "marginTop": 40.019999999999996,
            }
          }
        >
          <View
            style={
              {
                "backgroundColor": "#f59042",
                "borderRadius": 7.5,
                "height": 15,
                "width": 15,
              }
            }
          />
          <Text
            allowFontScaling={false}
            style={
              [
                {
                  "alignSelf": "flex-start",
                  "color": "#FFFFFF",
                  "fontFamily": "Montserrat-Light",
                  "fontSize": 20.009999999999998,
                },
                {
                  "marginLeft": 15,
                },
              ]
            }
          >
            Today  08:30
          </Text>
        </View>
        <Image
          source={
            {
              "testUri": "../../../src/assets/icons/clouds.png",
            }
          }
          style={
            {
              "height": 200.1,
              "tintColor": "#FFFFFF",
              "width": 225,
            }
          }
        />
        <Text
          allowFontScaling={false}
          style={
            [
              {
                "alignSelf": "flex-start",
                "color": "#FFFFFF",
                "fontFamily": "Montserrat-Bold",
                "fontSize": 20.009999999999998,
              },
              {
                "alignSelf": "center",
                "fontSize": 40.019999999999996,
              },
            ]
          }
        >
          NaN Â°C
        </Text>
        <View
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#28356b",
              "borderRadius": 22.5,
              "justifyContent": "center",
              "marginTop": 26.68,
              "maxWidth": 277.5,
              "paddingHorizontal": 22.5,
              "paddingVertical": 13.34,
            }
          }
        >
          <Text
            allowFontScaling={false}
            numberOfLines={1}
            style={
              [
                {
                  "color": "#FFFFFF",
                  "fontFamily": "Montserrat-Regular",
                  "fontSize": 20.009999999999998,
                },
                undefined,
              ]
            }
          />
        </View>
        <View
          style={
            [
              {
                "alignItems": "center",
                "flexDirection": "row",
                "marginTop": 40.019999999999996,
              },
              {
                "width": "100%",
              },
            ]
          }
        >
          <View
            style={
              {
                "marginRight": "1%",
                "marginTop": 26.68,
                "width": "32%",
              }
            }
          >
            <Image
              source={
                {
                  "testUri": "../../../src/assets/icons/humidity.png",
                }
              }
              style={
                {
                  "height": 53.36,
                  "resizeMode": "contain",
                  "tintColor": "#FFFFFF",
                  "width": "100%",
                }
              }
            />
            <Text
              allowFontScaling={false}
              style={
                [
                  {
                    "alignSelf": "flex-start",
                    "color": "#FFFFFF",
                    "fontFamily": "Montserrat-Bold",
                    "fontSize": 20.009999999999998,
                  },
                  {
                    "alignSelf": "center",
                    "marginTop": 13.34,
                  },
                ]
              }
            >
              undefined%
            </Text>
            <Text
              allowFontScaling={false}
              numberOfLines={1}
              style={
                [
                  {
                    "color": "#FFFFFF",
                    "fontFamily": "Montserrat-Regular",
                    "fontSize": 20.009999999999998,
                  },
                  {
                    "alignSelf": "center",
                    "marginTop": 13.34,
                  },
                ]
              }
            >
              Humidity
            </Text>
          </View>
          <View
            style={
              {
                "marginRight": "1%",
                "marginTop": 26.68,
                "width": "32%",
              }
            }
          >
            <Image
              source={
                {
                  "testUri": "../../../src/assets/icons/temprature.png",
                }
              }
              style={
                {
                  "height": 53.36,
                  "resizeMode": "contain",
                  "tintColor": "#FFFFFF",
                  "width": "100%",
                }
              }
            />
            <Text
              allowFontScaling={false}
              style={
                [
                  {
                    "alignSelf": "flex-start",
                    "color": "#FFFFFF",
                    "fontFamily": "Montserrat-Bold",
                    "fontSize": 20.009999999999998,
                  },
                  {
                    "alignSelf": "center",
                    "marginTop": 13.34,
                  },
                ]
              }
            >
              NaNÂ°C
            </Text>
            <Text
              allowFontScaling={false}
              numberOfLines={1}
              style={
                [
                  {
                    "color": "#FFFFFF",
                    "fontFamily": "Montserrat-Regular",
                    "fontSize": 20.009999999999998,
                  },
                  {
                    "alignSelf": "center",
                    "marginTop": 13.34,
                  },
                ]
              }
            >
              Feels Like
            </Text>
          </View>
          <View
            style={
              {
                "marginRight": "1%",
                "marginTop": 26.68,
                "width": "32%",
              }
            }
          >
            <Image
              source={
                {
                  "testUri": "../../../src/assets/icons/wind.png",
                }
              }
              style={
                {
                  "height": 53.36,
                  "resizeMode": "contain",
                  "tintColor": "#FFFFFF",
                  "width": "100%",
                }
              }
            />
            <Text
              allowFontScaling={false}
              style={
                [
                  {
                    "alignSelf": "flex-start",
                    "color": "#FFFFFF",
                    "fontFamily": "Montserrat-Bold",
                    "fontSize": 20.009999999999998,
                  },
                  {
                    "alignSelf": "center",
                    "marginTop": 13.34,
                  },
                ]
              }
            >
              undefinedm/s
            </Text>
            <Text
              allowFontScaling={false}
              numberOfLines={1}
              style={
                [
                  {
                    "color": "#FFFFFF",
                    "fontFamily": "Montserrat-Regular",
                    "fontSize": 20.009999999999998,
                  },
                  {
                    "alignSelf": "center",
                    "marginTop": 13.34,
                  },
                ]
              }
            >
              Wind Speed
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
